<?xml version="1.0"?>
<launch>
  <!-- start Webots -->
  <arg name="no-gui" default="false" doc="Start Webots with minimal GUI"/>

  <include file="$(find webots_ros)/launch/webots.launch">
    <arg name="mode" value="realtime"/>
    <arg name="no-gui" value="$(arg no-gui)"/>
    <arg name="world" value="$(env HOME)/projects/webots/wacker_vision/worlds/first_experiment.wbt"/>
  </include>

  <arg name="camera_name" default="camera" />
  <!-- frame_id for the camera -->
  <arg name="frame_id" default="camera_link" />
  <!-- By default, calibrations are stored to file://${ROS_HOME}/camera_info/${NAME}.yaml
   To use your own fill this arg with the corresponding url, e.g.:
   "file:///$(find your_camera_package)/config/your_camera.yaml" -->
  <arg name="camera_info_url" default="file:///$(find webots_arm)/config/webots_wacker_cam.yaml" />

  <!-- <arg name="node_start_delay" default="1.0" />
  <node name="listener" pkg="roscpp_tutorials" type="listener" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' " /> -->

  <!-- <group ns="$(arg camera_name)"> -->
    <node name="camera_control" pkg="webots_arm" type="camera_control" output="screen" required="true">
      <param name="frame_id" type="string" value="$(arg frame_id)" />
      <param name="camera_name" type="string" value="$(arg camera_name)" />
      <param name="camera_info_url" type="string" value="$(arg camera_info_url)" />
    </node>
  <!-- </group> -->
</launch>
